### PR TITLE
Extract generic DDS fuzz model from interval collections fuzz tests

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

This separates the portions of the interval collection fuzz tests which are DDS-specific from those that are SharedString/interval collection specific. The goal of this refactor is to enable moving the generic DDS harness to a shared location (I'm currently thinking `@fluid-internal/test-dds-utils` is a good fit) to reduce boilerplate in other DDS's fuzz models.

It also includes some mild refactors & feature additions to stochastic-test-utils, which were implemented as needed (or as I found the current API overly restrictive).

### CODE REVIEW

File: packages/test/stochastic-test-utils/src/generators.ts [hunk 1]
```diff
@@ -148,27 +149,39 @@ export function chainIterables<T, TState>(
...
+ * Exhausts both input generators before terminating by default. If {@link ExitBehavior.OnEitherExhausted} is
+ * provided, instead exits as soon as the next element it would produce is `done`.
  *
  * @example
  * ```typescript
  * // Assume gen1 produces 1, 2, 3, ... and gen2 produces "a", "b", "c", ...
  * interleave(gen1, gen2) // 1, a, 2, b, 3, c, ...
  * interleave(gen1, gen2, 2) // 1, 2, a, 3, 4, b, 5, 6, c, ...
- * interleave(gen1, gen2, 2, 3) // 1, 2, a, b, c, 3, 4, d, e, f, ...
+ * interleave(take(2, gen1), gen2, 1, 1, ExitBehavior.OnEitherExhausted) // 1, a, 2, b
```
Comment (scarlettjlee) [inline packages/test/stochastic-test-utils/src/generators.ts:174]: Not a big deal since the description above is accurate, but the example in the test is clearer to me that it's not this:
interleave(gen1, take(2, gen2), 1, 1, ExitBehavior.OnEitherExhausted) // 1, a, 2, b
but this:
interleave(gen1, take(2, gen2), 1, 1, ExitBehavior.OnEitherExhausted) // 1, a, 2, b, 3

File: packages/dds/sequence/src/test/intervalCollection.fuzz.spec.ts [hunk 2]
```diff
@@ -147,22 +118,19 @@ function makeOperationGenerator(
 	}
 
 	// All subsequent helper functions are generators; note that they don't actually apply any operations.
-	function startPosition({ random, sharedString }: ClientOpState): number {
-		return random.integer(0, Math.max(0, sharedString.getLength() - 1));
+	function startPosition({ random, channel }: ClientOpState): number {
```
Comment (scarlettjlee) [inline packages/dds/sequence/src/test/intervalCollection.fuzz.spec.ts:121]: Why the change in name to "channel" since it's not used like a channel?

Comment (Abe27342) [reply packages/dds/sequence/src/test/intervalCollection.fuzz.spec.ts:121]: the prop comes from the generic infrastructure, so it shouldn't be `sharedString`. we could rename the local to `sharedString` if we want. naming the field `dds` would also be reasonable and I considered it. if you think that's better I'm open to changing

File: packages/dds/sequence/src/test/ddsFuzzHarness.ts [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+	if (!directory) {
+		return undefined;
+	}
+	const filepath = path.join(directory, `${seed}.json`);
+	return { saveOnFailure: true, filepath };
+}
+
+/**
+ * @example
+ * Typical
```
Comment (scarlettjlee) [inline packages/dds/sequence/src/test/ddsFuzzHarness.ts:91]: Did you mean to write more here or is this some shorthand I'm not familiar with?

Comment (Abe27342) [reply packages/dds/sequence/src/test/ddsFuzzHarness.ts:91]: lol, yeah. definitely unfinished thought. thanks :)

**Review by scarlettjlee — APPROVED** [review]

File: packages/dds/sequence/src/test/intervalCollection.fuzz.spec.ts (hunk 2 above)
Comment (scarlettjlee) [reply packages/dds/sequence/src/test/intervalCollection.fuzz.spec.ts:121]: I wrote the comment before I traced where it came from, and then deleted the comment. `dds` isn't really better than `channel` currently. (I have heard about making data stores also channels, but I'm going to disregard that.)
